projects/demo/src/modules/app/classes/ts-file.parser.ts

```diff
@@ -1,53 +1,61 @@
+import {TuiDocCodeProcessedValue} from '@taiga-ui/addon-doc';
+
 export class TsFileParser {
     get className(): string {
-        const [, className] = this.rawFileContent.match(/(?:export class\s)(\w*)/i) || [];
+        const [, className] =
+            this.rawFileContent?.match(/(?:export class\s)(\w*)/i) ?? [];
 
-        return className || '';
+        return className ?? '';
     }
 
     set className(newClassName: string) {
-        this.rawFileContent = this.rawFileContent.replace(
+        this.rawFileContent = this.rawFileContent?.replace(
             /(export class\s)(\w*)/i,
             '$1' + newClassName,
         );
     }
 
     get hasNgModule(): boolean {
-        return this.rawFileContent.includes('@NgModule');
+        return this.rawFileContent?.includes('@NgModule') ?? false;
     }
 
     get hasNgComponent(): boolean {
-        return this.rawFileContent.includes('@Component');
+        return this.rawFileContent?.includes('@Component') ?? false;
     }
 
-    constructor(protected rawFileContent: string) {
-        const classesInside = rawFileContent.match(/export class/gi) || [];
+    constructor(protected rawFileContent: TuiDocCodeProcessedValue) {
```

Comment:
Let's turn `rawFileContent` into the string, by default

or dont create these class if `TuiDocCodeProcessedValue` is not string

Reply:
So much optional chaining (in this class and its descendants) scares me :(